tests/test_optics.py

```diff
@@ -570,13 +543,20 @@ def test_OpticalPSF_pupil_plane():
     ref_psf = galsim.OpticalPSF(lam_over_diam, obscuration=obscuration, nstruts=nstruts,
                                 oversampling=pp_oversampling, strut_angle=strut_angle,
                                 pad_factor=pp_pad_factor)
-    im = galsim.fits.read(os.path.join(imgdir, pp_file))
+    if os.path.isfile(os.path.join(imgdir, pp_file)):
+        im = galsim.fits.read(os.path.join(imgdir, pp_file))
+    else:
+        im = galsim.Image(ref_psf._psf.aper.illuminated.astype(float))
+        im.write(os.path.join(imgdir, pp_file))
+    ref_im_scale = ref_psf._psf.aper.pupil_plane_scale
     test_psf = galsim.OpticalPSF(lam_over_diam, obscuration=obscuration,
                                  oversampling=pp_oversampling, pupil_plane_im=im,
-                                 pad_factor=pp_pad_factor)
+                                 pad_factor=pp_pad_factor,
+                                 _pupil_plane_scale=ref_im_scale)
```

Comment:
@cpadavis and I recently started trying to use the DECam pupil plane image with OpticalPSF, and we found the API to be rather counter-intuitive, since it seems like you have to magically know what scale size GalSim will decide to set for the wavefront image and match the pupil plane image to that. 

I was thinking that it would make more sense to let the user specify a pixel_scale for the pupil plane image by hand.  i.e. not as a parameter (as here) but via the `scale` parameter in the image that you provide. 

This prescription of matching the image to the telescope diameter seems ok (certainly better than the current version), but maybe we can just do that when there is no `image.scale` set?  And add some docs about what units the pupil plane image scale should be in, so if they know it for their image, they can just provide it directly.

I mentioned to Chris that you'll be up here at SLAC tomorrow, so maybe we discuss in person some time tomorrow or Friday.